### PR TITLE
doc(rest api): Update REST API extension documentation to cover Community Edition

### PR DESCRIPTION
**Pull request by amottier**
Add information about creation of REST API extension using Community Edition.

Closes BS-17850

### CODE REVIEW

File: md/rest-api-extensions.md [hunk 1]
```diff
@@ -1,37 +1,197 @@
...
+        }
+
+        // Prepare the result
+        def result = [p: p, c: c, userInformation: usersInformation]
+
+        int startIndex = p*c
+        int endIndex = p*c + users.size() - 1
+
+        // Send the result as a JSON representation
+        return buildPagedResponse(responseBuilder, new JsonBuilder(result).toPrettyString(), startIndex, endIndex, context.apiClient.identityAPI.numberOfUsers)
```
Comment (tbouffard) [inline md/rest-api-extensions.md]:
```suggestion
        return buildPagedResponse(responseBuilder, new JsonBuilder(result).toString(), startIndex, endIndex, context.apiClient.identityAPI.numberOfUsers)
```

Comment (tbouffard) [reply md/rest-api-extensions.md]: ℹ️ This is to reduce the response size and the same should be applied for the other example (see line 248)

File: md/rest-api-extensions.md [hunk 2]
```diff
@@ -1,37 +1,197 @@
...
 ## Prerequisites
 
 Prerequisites for developing a REST API extension are:
 
 * Java/Groovy development expertise.
-* Basic knowledge of Maven.
+
+## Example description
+
+The following sections show how to create a REST API extension. As an example, we create a REST API extension that use Bonita BPM Engine to provide user information (first name, last name, email address).
```
Comment (tbouffard) [inline md/rest-api-extensions.md]: ⚠️  typos

Comment (tbouffard) [inline md/rest-api-extensions.md]:
```suggestion
The following sections show how to create a REST API extension. As an example, we create a REST API extension that uses the Bonita BPM Engine to provide user information (first name, last name, email address).
```

File: md/rest-api-extensions.md [hunk 3]
```diff
@@ -1,37 +1,197 @@
...
+
+## For Community Edition
+
+### Create a new REST API extension skeleton
+
+A single REST API extension can define several API extensions.
+
+A REST API extension includes two different type of files:
+* One Groovy script files per API extension. The Groovy script files define the business logic of the API extensions.
```
Comment (tbouffard) [inline md/rest-api-extensions.md]: ⚠️ typo

Comment (tbouffard) [inline md/rest-api-extensions.md]:
```suggestion
* One Groovy script file per API extension. The Groovy script files define the business logic of the API extensions.
```

**Review by tbouffard — CHANGES_REQUESTED** [review]
Good job, very clear.
Minor changes requested and suggested

**Comment by amottier** [comment]
@tbouffard I amend my commit with your recommendation. If everything is ok can you do a quick final review and merge? Thanks.

**Comment by tbouffard** [comment]
LGTM, for the next time, could you avoid force-pushing, this makes the review harder.
We can squash commit while merging the PR 😸

**Review by tbouffard — APPROVED** [review]

**Pull request merged**